### PR TITLE
moved recommendations after application scenarios. 

### PR DESCRIPTION
**Pull request by ms-s**
explain how tls exporter helps with KDF and mitigating selfie attacks. added text on psk identity usage.

### CODE REVIEW

File: draft-dt-tls-external-psk-guidance.md
Comment (jhoyla) [inline draft-dt-tls-external-psk-guidance.md]: This feels like a bit of a non-sequitur. We're not using PSK importers to change the hash-function, we using them to make the keys distinct.

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: This bullet is not something that can "vary depending on the deployment." Can we just move this to its own paragraph below the list, or merge with the first paragraph in this section?

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: The second and third sentences in this bullet seem like too much information for the requirement. Can we please revert to the previous text?

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: This is redundant with the text in I-D.ietf-tls-external-psk-importer. Since item (3) already recommends using it, I think we should remove this entire bullet.

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: It's really hard for me to pull out the requirement from this paragraph. Can we please simplify it, or remove it altogether?

**Review by chris-wood — CHANGES_REQUESTED** [review]

File: draft-dt-tls-external-psk-guidance.md
Comment (jhoyla) [inline draft-dt-tls-external-psk-guidance.md]: Maybe we should drop a link to RFC 4122 (UUID) here? Can you create stable ID from a UUID? Do we need to consider access to clock?

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: This was done based on feedback from Björn. Feel free to move around this text to some other suitable place in the document. But I agree with Björn that this is definitely worth highlighting.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: Added a reference to RFC 4122. Thanks for the suggestion.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: That's correct. Moved this above the bullet list.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: This is a guidance document so it is worth pointing out different ways of doing things. I added a reference to UUID (RFC 4122) based on feedback from Jonathan. I understand that you want to keep the recommendations concise. So moved this text to the security considerations.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: It is quite simple. We don't recommend sharing PSKs among many nodes. If someone's use case still wants that, then they should use external PSK importers. However, that by itself is not enough. Nodes should know the other end points identifier to prevent rerouting of attacks. And of course, endpoints that know the PSK can always impersonate.

Comment (chris-wood) [reply draft-dt-tls-external-psk-guidance.md]: This is a lot of text to simply say, "If you share PSKs, make sure you also include some other form of unique identifier." Can we please simplify it?

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md:192]: Please revert deleting this text!

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: This paragraph does not add information. Please remove it.

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md:298]: nit: extra whitespace

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md:264]: nit: extra whitespace

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: nit: extra whitespace

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md]: As in my other comment, I think this is too much text to get the point across. It needs to be simplified

**Review by chris-wood — CHANGES_REQUESTED** [review]

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md:192]: I am really sorry for the accidental deletion. Reverted.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: With Björn's agreement, this is now removed.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md:298]: Should be fixed.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md:264]: Should be fixed now!

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: I have shortened the security considerations section somewhat. We can further shorten it if others also feel the same way.

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [reply draft-dt-tls-external-psk-guidance.md:192]: No problem!

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md:179]: This whole section section seems to have different formatting. I'd prefer we just revert this entirely. (That is, break lines at ~80-120 columns, give or take.) I can do that after we merge though.

File: draft-dt-tls-external-psk-guidance.md
Comment (chris-wood) [inline draft-dt-tls-external-psk-guidance.md:300]: There's nothing in this now-empty section?

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md:179]: Yes. My editor doesn't add line breaks and I seem to like it. I guess it is best if you do it after the merge.

File: draft-dt-tls-external-psk-guidance.md
Comment (ms-s) [reply draft-dt-tls-external-psk-guidance.md]: @chris-wood made another commit that further simplifies the text.

**Review by chris-wood — APPROVED** [review]
LGTM pending one orphan section!

**Comment by ms-s** [comment]
I left it empty for Owen (@upros). He was planning to contribute some text on PSK collisions (between external PSKs and between external and internal PSKs).

**Comment by chris-wood** [comment]
I'll merge and will remove it for now. Owen can add it when he gets around to his PR.

**Pull request merged**
